core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -689,6 +689,58 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     new ParallelCollectionRDD[T](this, seq, numSlices, Map[Int, Seq[String]]())
   }
 
+  /**
+   * Creates a new RDD[Long] containing elements from `start` to `end`(exclusive), increased by
+   * `step` every element.
+   *
+   * @note if we need to cache this RDD, we should make sure each partition does not exceed limit.
+   *
+   * @param start the start value.
+   * @param end the end value.
+   * @param step the incremental step
+   * @param numSlices the partition number of the new RDD.
+   * @return
+   */
+  def range(
+      start: Long,
+      end: Long,
+      step: Long = 1,
+      numSlices: Int = defaultParallelism): RDD[Long] = withScope {
+    assertNotStopped()
+    // when step is 0, range will run infinitely
+    require(step != 0, "step cannot be 0")
+    val numElements: BigInt = {
+      val safeStart = BigInt(start)
+      val safeEnd = BigInt(end)
+      if (((safeEnd - safeStart) % step).toInt == 0) {
+        (safeEnd - safeStart) / step
+      } else {
+        (safeEnd - safeStart) / step + 1
+      }
+    }
+    parallelize(0 until numSlices, numSlices).mapPartitionsWithIndex((i, _) => {
+      val partitionStart = ((i * numElements) / numSlices * step + start).toLong
+      val partitionEnd = (((i + 1) * numElements) / numSlices * step + start).toLong
```

Comment:
As our previous discussion, `partitionEnd` may exceed `end`. If `end` is `Long.MaxValue`, `partitionEnd` may overflow. Could you compare it with `Long.MaxValue` before converting it to `Long`, and set it to `Long.MaxValue` if `partitionEnd` overflow?